packages/replay/src/coreHandlers/handleAfterSendEvent.ts

```diff
@@ -0,0 +1,87 @@
+import { getCurrentHub } from '@sentry/core';
+import type { Event, Transport, TransportMakeRequestResponse } from '@sentry/types';
+
+import { UNABLE_TO_SEND_REPLAY } from '../constants';
+import type { ReplayContainer } from '../types';
+import { isErrorEvent, isTransactionEvent } from '../util/eventUtils';
+
+type AfterSendEventCallback = (event: Event, sendResponse: TransportMakeRequestResponse | void) => void;
+
+/**
+ * Returns a listener to be added to `client.on('afterSendErrorEvent, listener)`.
+ */
+export function handleAfterSendEvent(replay: ReplayContainer): AfterSendEventCallback {
+  // Custom transports may still be returning `Promise<void>`, which means we cannot expect the status code to be available there
+  const enforceStatusCode = isBaseTransportSend();
```

Comment:
l: just to remind ourselves: 

```suggestion
  // Custom transports may still be returning `Promise<void>`, which means we cannot expect the status code to be available there
  // TODO (v8): remove this check as it will no longer be necessary
  const enforceStatusCode = isBaseTransportSend();
```